plugins/spectate/spectate.cpp

```diff
@@ -36,51 +46,300 @@ using namespace DFHack;
 using namespace Pausing;
 using namespace df::enums;
 
-void onTick(color_ostream& out, void* tick);
-void onJobStart(color_ostream &out, void* job);
-void onJobCompletion(color_ostream &out, void* job);
+struct Configuration {
+    bool debug = false;
+
+    bool unpause = false;
+    bool disengage = false;
+    bool animals = false;
+    bool hostiles = true;
+    bool visitors = false;
+    int32_t tick_threshold = 1000;
+} config;
 
-uint64_t tick_threshold = 1000;
-bool focus_jobs_enabled = false;
-bool disengage_enabled = false;
-bool unpause_enabled = false;
 Pausing::AnnouncementLock* pause_lock = nullptr;
 bool lock_collision = false;
 bool announcements_disabled = false;
 
-bool following_dwarf = false;
-df::unit* our_dorf = nullptr;
-df::job* job_watched = nullptr;
-int32_t timestamp = -1;
-
-std::set<int32_t> job_tracker;
-std::map<uint16_t,int16_t> freq;
-std::default_random_engine RNG;
-
-void enable_auto_unpause(color_ostream &out, bool state);
-
-
 #define base 0.99
 
 static const std::string CONFIG_KEY = std::string(plugin_name) + "/config";
 enum ConfigData {
     UNPAUSE,
     DISENGAGE,
-    JOB_FOCUS,
-    TICK_THRESHOLD
+    TICK_THRESHOLD,
+    ANIMALS,
+    HOSTILES,
+    VISITORS
+
 };
 
-static PersistentDataItem config;
-inline void saveConfig() {
-    if (config.isValid()) {
-        config.ival(UNPAUSE) = unpause_enabled;
-        config.ival(DISENGAGE) = disengage_enabled;
-        config.ival(JOB_FOCUS) = focus_jobs_enabled;
-        config.ival(TICK_THRESHOLD) = tick_threshold;
-    }
-}
+static PersistentDataItem pconfig;
 
+DFhackCExport command_result plugin_enable(color_ostream &out, bool enable);
 command_result spectate (color_ostream &out, std::vector <std::string> & parameters);
+#define COORDARGS(id) id.x, (id).y, id.z
```

Comment:
any reason there are params around the y and not the others?

Reply:
typo